components/learn/modules/ROOT/pages/unit-testing.adoc

```diff
@@ -0,0 +1,126 @@
+= Testing Smart Contracts
+
+In a blockchain environment, a single mistake could cost you all of your funds (or even worse, your users'!). This guide will teach you how to write automated tests that verify your application behaves exactly as you intended.
+
+== About Testing
+
+If you ever wrote a program, you also run into to see what happens. That was a test. But not a great one, because you then had to re-run your test manually for every single change to your code. Enter automated testing.
```

Comment:
```suggestion
If you ever wrote a program, you also ran it to see what happens. That was a test. But not a great one, because you then had to re-run your test manually for every single change to your code. Enter automated testing.
```